overlord/snapstate/handlers.go

```diff
@@ -1777,6 +1777,10 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		return err
```

Comment:
the changes in snapstate need some unit tests

Reply:
I added a test.